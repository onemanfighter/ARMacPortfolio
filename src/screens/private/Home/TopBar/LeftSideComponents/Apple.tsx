import { AppleIcon } from '@assets';
import { Menu, MenuDivider } from '@chakra-ui/react';
import {
  MenuItemComponent,
  MenuListComponent,
  TopBarButton,
} from '@components';
import {
  activeAppActionsSelector,
  activeAppSelector,
  loginSelector,
  processStore,
} from '@processStore';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
  usersSelector,
} from '@settingsStore';
import { useTranslation } from 'react-i18next';

const Apple = () => {
  const { t } = useTranslation();
  const { iconColor } = settingsStore(useShallow(darkModeColorSelector));
  const { name } = settingsStore(useShallow(usersSelector)).userData;
  const { lockUser, logout, clearAllActiveApps } = processStore((state) => ({
    ...loginSelector(state),
    ...activeAppActionsSelector(state),
  }));

  return (
    <Menu>
      <TopBarButton
        text=""
        onClick={() => {}}
        ariaLabel="apple-top-bar-button"
        icon={<AppleIcon width="1.5em" height="1.5em" color={iconColor} />}
      />
      <MenuListComponent>
        <MenuItemComponent
          text={t('TopAppBar.apple.aboutThisMac')}
          ariaLabel="about-this-mac"
          onClick={() => {}}
          side="left"
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.apple.systemSettings')}
          ariaLabel="system-settings"
          onClick={() => {}}
          side="left"
        />
        <MenuItemComponent
          text={t('TopAppBar.apple.appStore')}
          ariaLabel="app-store"
          onClick={() => {}}
          side="left"
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.apple.recentItems')}
          ariaLabel="recent-items"
          onClick={() => {}}
          command=">"
          side="left"
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.apple.forceQuit')}
          ariaLabel="force-quit"
          onClick={() => {}}
          side="left"
          command="⌥⌘⎋"
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.apple.sleep')}
          ariaLabel="sleep"
          side="left"
          onClick={() => {}}
        />
        <MenuItemComponent
          text={t('TopAppBar.apple.restart')}
          ariaLabel="restart"
          side="left"
          onClick={() => {}}
        />
        <MenuItemComponent
          text={t('TopAppBar.apple.shutDown')}
          ariaLabel="shutDown"
          side="left"
          onClick={() => {}}
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.apple.lockScreen')}
          ariaLabel="lock-screen"
          side="left"
          onClick={() => {
            setTimeout(() => {
              lockUser();
            }, 500);
          }}
          command="⌃⇧Q"
        />
        <MenuItemComponent
          text={t('TopAppBar.apple.logOut', { userName: name })}
          ariaLabel="logout"
          side="left"
          onClick={() => {
            clearAllActiveApps();
            setTimeout(() => {
              logout();
            }, 1000);
          }}
          command="⌘⇧Q"
        />
      </MenuListComponent>
    </Menu>
  );
};

export default Apple;
