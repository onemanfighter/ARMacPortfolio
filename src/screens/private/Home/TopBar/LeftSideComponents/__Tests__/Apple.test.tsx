import { fireEvent, render, screen } from '@testing-library/react';
import Apple from '../Apple';

describe('Apple', () => {
  it('Should render apple button correctly', () => {
    const { container } = render(<Apple />);

    expect(container).toMatchSnapshot();
  });

  it('Should render apple menu correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));

    expect(container).toMatchSnapshot();
  });

  it('Should render about this mac correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('about-this-mac'));

    expect(container).toMatchSnapshot();
  });

  it('Should render system settings correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('system-settings'));

    expect(container).toMatchSnapshot();
  });

  it('Should render app store correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('app-store'));

    expect(container).toMatchSnapshot();
  });

  it('Should render recent items correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('recent-items'));

    expect(container).toMatchSnapshot();
  });

  it('Should render force quit correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('force-quit'));

    expect(container).toMatchSnapshot();
  });

  it('Should render sleep correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('sleep'));

    expect(container).toMatchSnapshot();
  });

  it('Should render restart correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('restart'));

    expect(container).toMatchSnapshot();
  });

  it('Should render shut down correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('shutDown'));

    expect(container).toMatchSnapshot();
  });

  it('Should render lock screen correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('lock-screen'));

    expect(container).toMatchSnapshot();
  });

  it('Should render log out correctly', () => {
    const { container } = render(<Apple />);

    fireEvent.click(screen.getByLabelText('apple-top-bar-button'));
    fireEvent.click(screen.getByLabelText('logout'));

    expect(container).toMatchSnapshot();
  });
});
