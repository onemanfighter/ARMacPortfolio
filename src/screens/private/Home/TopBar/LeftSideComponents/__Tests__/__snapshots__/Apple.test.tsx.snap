// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apple Should render apple button correctly 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="apple-top-bar-button"
    class="chakra-menu__menu-button css-bracgg"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <svg
        color="white"
        fill="currentColor"
        height="1.5em"
        viewBox="0 0 1024 1024"
        width="1.5em"
      >
        <path
          d="M747.4 535.7c-.4-68.2 30.5-119.6 92.9-157.5-34.9-50-87.7-77.5-157.3-82.8-65.9-5.2-138 38.4-164.4 38.4-27.9 0-91.7-36.6-141.9-36.6C273.1 298.8 163 379.8 163 544.6c0 48.7 8.9 99 26.7 150.8 23.8 68.2 109.6 235.3 199.1 232.6 46.8-1.1 79.9-33.2 140.8-33.2 59.1 0 89.7 33.2 141.9 33.2 90.3-1.3 167.9-153.2 190.5-221.6-121.1-57.1-114.6-167.2-114.6-170.7zm-105.1-305c50.7-60.2 46.1-115 44.6-134.7-44.8 2.6-96.6 30.5-126.1 64.8-32.5 36.8-51.6 82.3-47.5 133.6 48.4 3.7 92.6-21.2 129-63.7z"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-wkqcnc"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="about-this-mac"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="0"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        About This Mac
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="system-settings"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="1"
        id="menu-list-:r1:-menuitem-:r3:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        System Settings...
      </button>
      <button
        aria-label="app-store"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="2"
        id="menu-list-:r1:-menuitem-:r4:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        App Store...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="recent-items"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="3"
        id="menu-list-:r1:-menuitem-:r5:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          Recent Items
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          &gt;
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="force-quit"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="4"
        id="menu-list-:r1:-menuitem-:r6:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Force Quit...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="sleep"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="5"
        id="menu-list-:r1:-menuitem-:r7:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Sleep
      </button>
      <button
        aria-label="restart"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="6"
        id="menu-list-:r1:-menuitem-:r8:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Restart...
      </button>
      <button
        aria-label="shutDown"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="7"
        id="menu-list-:r1:-menuitem-:r9:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Shut Down...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="lock-screen"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="8"
        id="menu-list-:r1:-menuitem-:ra:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Lock Screen
      </button>
      <button
        aria-label="logout"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="9"
        id="menu-list-:r1:-menuitem-:rb:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Log out {{userName: string}}...
      </button>
    </div>
  </div>
</div>
`;

exports[`Apple Should render apple menu correctly 1`] = `
<div>
  <button
    aria-controls="menu-list-:rd:"
    aria-expanded="true"
    aria-haspopup="menu"
    aria-label="apple-top-bar-button"
    class="chakra-menu__menu-button css-bracgg"
    data-active=""
    id="menu-button-:rd:"
  >
    <span
      class="css-xl71ch"
    >
      <svg
        color="white"
        fill="currentColor"
        height="1.5em"
        viewBox="0 0 1024 1024"
        width="1.5em"
      >
        <path
          d="M747.4 535.7c-.4-68.2 30.5-119.6 92.9-157.5-34.9-50-87.7-77.5-157.3-82.8-65.9-5.2-138 38.4-164.4 38.4-27.9 0-91.7-36.6-141.9-36.6C273.1 298.8 163 379.8 163 544.6c0 48.7 8.9 99 26.7 150.8 23.8 68.2 109.6 235.3 199.1 232.6 46.8-1.1 79.9-33.2 140.8-33.2 59.1 0 89.7 33.2 141.9 33.2 90.3-1.3 167.9-153.2 190.5-221.6-121.1-57.1-114.6-167.2-114.6-170.7zm-105.1-305c50.7-60.2 46.1-115 44.6-134.7-44.8 2.6-96.6 30.5-126.1 64.8-32.5 36.8-51.6 82.3-47.5 133.6 48.4 3.7 92.6-21.2 129-63.7z"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    data-popper-escaped=""
    data-popper-placement="bottom-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-wkqcnc"
      id="menu-list-:rd:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="about-this-mac"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="0"
        id="menu-list-:rd:-menuitem-:re:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        About This Mac
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="system-settings"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="1"
        id="menu-list-:rd:-menuitem-:rf:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        System Settings...
      </button>
      <button
        aria-label="app-store"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="2"
        id="menu-list-:rd:-menuitem-:rg:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        App Store...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="recent-items"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="3"
        id="menu-list-:rd:-menuitem-:rh:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          Recent Items
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          &gt;
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="force-quit"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="4"
        id="menu-list-:rd:-menuitem-:ri:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Force Quit...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="sleep"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="5"
        id="menu-list-:rd:-menuitem-:rj:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Sleep
      </button>
      <button
        aria-label="restart"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="6"
        id="menu-list-:rd:-menuitem-:rk:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Restart...
      </button>
      <button
        aria-label="shutDown"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="7"
        id="menu-list-:rd:-menuitem-:rl:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Shut Down...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="lock-screen"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="8"
        id="menu-list-:rd:-menuitem-:rm:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Lock Screen
      </button>
      <button
        aria-label="logout"
        class="chakra-menu__menuitem css-s7lhnt"
        data-index="9"
        id="menu-list-:rd:-menuitem-:rn:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Log out {{userName: string}}...
      </button>
    </div>
  </div>
</div>
`;
